notifications/src/main/java/dk/shape/games/notifications/presentation/NotificationSettingsFragment.kt

```diff
@@ -0,0 +1,254 @@
+                subjectInfo.subjectName
+            }?.mapNotNull { subjectInfo ->
+
+                subscriptions.find { subscription ->
+                    subscription.subjectId == subjectInfo.subjectId
+                }?.takeIf { subscription ->
+                    subscription.types.isNotEmpty()
+                }?.let { matchingSubscription ->
+                    val statsNotificationGroups = appConfig.toStatsNotifications(subjectInfo)
+
+                    statsNotificationGroups.find { notificationGroup ->
+                        notificationGroup.sportId == subjectInfo.sportId
+                    }?.let { matchingGroup ->
+                        NotificationsSettingsSubjectViewModel(
+                            name = subjectInfo.subjectName,
+                            subscription = matchingSubscription,
+                            notificationGroup = matchingGroup,
+                            onSubjectNotificationTypesClicked = { action ->
+                                config.onSubjectNotificationTypesClicked(this, action)
+                            },
+                            onSetNotifications = { notificationTypes, onError ->
+                                lifecycleScope.launch {
+                                    whenResumed {
+                                        withContext(Dispatchers.IO) {
+                                            subjectNotificationsInteractor.updateNotifications(
+                                                deviceId = deviceId,
+                                                subjectId = matchingSubscription.subjectId,
+                                                subjectType = matchingSubscription.subjectType,
+                                                notificationTypeIds = notificationTypes.toTypeIds(),
+                                                onError = onError
+                                            )
+                                        }
+                                    }
+                                }
+                            }
+                        )
+                    }
+                }
+            }
+        } ?: listOf()
+
+    private fun NotificationSettingsAction.getEventIds(onResult: (List<String>?) -> Unit) {
+        when {
+            betslipComponentUUID != null -> {
+                onResult(config.provideEventIdsForBetSlip())
+            }
+            openedFromMyGames -> {
+                config.provideEventIdsForUserBetsAsync(onResult)
+            }
+            else -> onResult(null)
+        }
+    }
+
+    private fun AppConfig.toStatsNotifications(info: SubjectInfo): List<AppConfig.SubjectNotificationGroup> =
+        when (info.subjectType) {
+            SubjectType.TEAMS -> teamSubjectNotifications
+            SubjectType.ATHLETES -> athleteSubjectNotifications
+            else -> throw IOException("Notifications are only supported for teams and athletes at the moment.")
```

Comment:
Maybe this should be an `IllegalArgumentException` instead?
```suggestion
            else -> throw IllegalArgumentException("Notifications are only supported for teams and athletes at the moment.")
```